src/com/mellanox/cicd/Matrix.groovy

```diff
@@ -225,34 +225,54 @@ def gen_image_map(config) {
         config.runs_on_dockers.each { item ->
 
             def dfile = item.clone()
-            dfile.arch = dfile.arch ?: arch
+            config.logger.debug("run on dockers item: " + dfile)
+
+            if (dfile.enable == null) {
+                dfile.enable = "true"
+                config.logger.debug("run on dockers item.enable: " + dfile.enable)
+            }
 
-            if (dfile.arch && dfile.arch != arch) {
-                config.logger.trace(3, "skipped conf: " + arch + " name: " + dfile.name)
-                return
+            if (dfile.enable == "auto") {
+                dfile.enable = '${' + dfile.name + '}'
+                config.logger.debug("run on dockers item.enable: " + dfile.enable)
             }
 
-            dfile.file = dfile.file ?: ''
-            if (dfile.url) {
-                parts = dfile.url.tokenize('/').last().tokenize(':')
-                if (parts.size() == 2) {
-                    dfile.tag = parts[1]
-                    tag_size = dfile.tag.size() + 1
-                    len = dfile.url.size() - tag_size
-                    dfile.uri = dfile.url.substring(0,len)
+            def enable = resolveTemplate(dfile, dfile.enable, config)
+
+            config.logger.debug("run on dockers enable: " + enable)
+
+            if (enable.toBoolean()) {
+                dfile.arch = dfile.arch ?: arch
```

Comment:
IMO arch set is too late, arch can be used as part of other vars and will be unavailable during ```resolveTemplate```

Reply:
I does not change existed code, I hide exist code under "if enable"